Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def actual_url(url)
+    m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
```

Comment:
@mikemcquaid Sorry, but your version is still incorrect. `-L` cannot be used with code recursion. 

I implement my own version which can be viewed from https://github.com/Homebrew/homebrew/compare/Homebrew:master...xu-cheng:curl